File: src/javakotlinexamples/java/LoopDemo.java
package javakotlinexamples.java;

public class LoopDemo {

    public static void main(String[] args) {
        for (int i = 1; i <= 3; i++) {
            System.out.println(i);
        }

        for (int i = 6; i >= 0; i -= 2) {
            System.out.println(i);
        }

        var array = new String [] {"Eins", "Zwei", "Drie"};
        for (var s : array) {
            System.out.println(s);
        }
    }
}

File: src/javakotlinexamples/kotlin/LoopDemo.kt
package javakotlinexamples.kotlin

fun main(args: Array<String>) {

    // for loop iterates through anything that provides an iterator.

    for (i in 1..3) {
        println(i)
    }

    for (i in 6 downTo 0 step 2) {
        println(i)
    }

    val array = arrayOf("Eins", "Zwei", "Drei")
    for (i in array.indices) {
        println(array[i])
    }
}